xcode/Subconscious/Shared/Components/OmniboxView.swift

```diff
@@ -71,29 +71,36 @@ struct OmniboxView: View {
                 .stroke(Color.separator, lineWidth: 0.5)
         )
         .overlay(
-            // Loading pulse
-            RoundedRectangle(cornerRadius: AppTheme.cornerRadius)
-                .stroke(
-                    .conicGradient(
-                        gradient,
-                        center: .center,
-                        angle: Angle.radians(phase)),
-                    lineWidth: 2
-                )
-                .animation(
-                    .linear(duration: Duration.loading * 2)
-                        .repeatForever(autoreverses: false),
-                    value: phase
-                )
-                // Keep the view mounted but invisible
-                .opacity(status == .loading ? 0.5 : 0)
+            VStack {
+                if status == .loading {
+                // Loading pulse
+                RoundedRectangle(cornerRadius: AppTheme.cornerRadius)
+                    .stroke(
+                        .conicGradient(
+                            gradient,
+                            center: .center,
+                            angle: Angle.radians(phase)),
+                        lineWidth: 2
+                    )
+                    .animation(
+                        .linear(duration: Duration.loading * 2)
+                            .repeatForever(autoreverses: false),
+                        value: phase
+                    )
+                    // Keep the view mounted but invisible
+                    .opacity(status == .loading ? 0.5 : 0)
+                    .task {
+                        if status == .loading {
+                            phase = Self.initialAngle + Self.fullRotation
+                        } 
+                    }
+                    .onDisappear() {
+                        phase = Self.initialAngle
+                    }
```

Comment:
Had to do some tweaks to make this support entering and leaving the loading state multiple times.